### PR TITLE
Fill news for JuMP v0.21

### PR DESCRIPTION
**Pull request by blegat**

### CODE REVIEW

File: NEWS.md [hunk 1]
```diff
@@ -1,6 +1,50 @@
...
+- The macros now create the containers with `map` instead of `for` loops,
+  as a consequence, containers created by `@expression` can now have any element
+  type and containers of constraint references now have concrete element types
+  when possible (#2070).
+- Calling `deepcopy(::AbstractModel)` now throws an error.
+- The constraint name is now printed in the model string (#2108).
+
+New features:
+
+- Added support for solver-independent and solver-specific (#2101).
```
Comment (odow) [inline NEWS.md]: missing "callbacks"

File: NEWS.md [hunk 2]
```diff
@@ -1,6 +1,50 @@
...
+- `SDconstraint(model, x >= 0)` is now equivalent to `constraint(model, x in PSDCone())`
+  instead of `constraint(model, (x .- 0) in PSDCone())` (#2107).
+- The macros now create the containers with `map` instead of `for` loops,
+  as a consequence, containers created by `@expression` can now have any element
+  type and containers of constraint references now have concrete element types
+  when possible (#2070).
+- Calling `deepcopy(::AbstractModel)` now throws an error.
+- The constraint name is now printed in the model string (#2108).
+
+New features:
```
Comment (mlubin) [inline NEWS.md]: `set_optimizer_attributes` is new also

File: NEWS.md [hunk 3]
```diff
@@ -1,6 +1,50 @@
...
+  * replace `with_optimizer(Ipopt.Optimizer, max_cpu_time=60.0)`
+    by `optimizer_with_attributes(Ipopt.Optimizer, "max_cpu_time" => 60.0)`.
+  * replace `with_optimizer(Gurobi.Optimizer, env)` by `() -> Gurobi.Optimizer(env)`.
+  * replace `with_optimizer(Gurobi.Optimizer, env, Presolve=0)`
+    by `optimizer_with_attributes(() -> Gurobi.Optimizer(env), "Presolve" => 0)`.
+
+  alternatively to `optimizer_with_attributes`, you can also set the attributes
+  separately with `set_optimizer_attribute`.
+- Renamed `set_parameter` to `set_optimizer_attribute` (#2150).
+- Broadcast should now be explicit inside macros. `SDconstraint(model, x >= 1)`
```
Comment (mlubin) [inline NEWS.md]: Typo: `@SDconstraint`

File: NEWS.md [hunk 4]
```diff
@@ -1,6 +1,50 @@
...
+  and `@constraint(model, x + 1 in SecondOrderCone())` now throw an error
+  instead of broadcasting `1` along the dimension of `x` (#2107).
+- `SDconstraint(model, x >= 0)` is now equivalent to `constraint(model, x in PSDCone())`
+  instead of `constraint(model, (x .- 0) in PSDCone())` (#2107).
+- The macros now create the containers with `map` instead of `for` loops,
+  as a consequence, containers created by `@expression` can now have any element
+  type and containers of constraint references now have concrete element types
+  when possible (#2070).
```
Comment (mlubin) [inline NEWS.md]: I would add more: This fixes a long-standing issue where `@expression` could only be used to generate a collection of linear expressions. Now it works for quadratic expressions as well.

File: NEWS.md [hunk 5]
```diff
@@ -1,6 +1,50 @@
...
+- Calling `deepcopy(::AbstractModel)` now throws an error.
+- The constraint name is now printed in the model string (#2108).
+
+New features:
+
+- Added support for solver-independent and solver-specific callbacks (#2101).
+- Added `write_to_file` and `read_from_file` (#2114).
```
Comment (mlubin) [inline NEWS.md]: Which file formats are supported?

Comment (odow) [reply NEWS.md]: CBF, LP, MathOptFormat, MPS, SDPA

File: NEWS.md [hunk 6]
```diff
@@ -1,6 +1,50 @@
...
+- Added support for complementarity constraints (#2132).
+- Added support for indicator @constraints (#2092).
```
Comment (dourouc05) [inline NEWS.md]: Why an @ here and not for the previous line?

File: NEWS.md [hunk 7]
```diff
@@ -1,6 +1,54 @@
 JuMP release notes
 ==================
 
+Version 0.21 (Feb 13, 2020)
+-----------------------------
+
+Breaking changes:
+
+- Deprecated `with_optimizer` (#2090, #2084, #2141). You can replace
```
Comment (mlubin) [inline NEWS.md]: We should provide a motivation for making this breaking change. I.e., in order to improve th econsistency of how solver parameters are specified...

Comment (mlubin) [reply NEWS.md]: Thinking about this again, it's not necessary. We link to the PRs that have these discussions.

**Pull request merged**